lib/Model/Pass/PurgeUnnamedAndUnreachableTypes.cpp

```diff
@@ -68,19 +138,18 @@ void model::purgeUnreachableTypes(TupleTree<model::Binary> &Model) {
 static void model::purgeTypesImpl(TupleTree<model::Binary> &Model,
                                   bool KeepTypesWithName) {
   struct NodeData {
-    model::Type *T;
+    const model::TypeDefinition *T;
   };
   using Node = ForwardNode<NodeData>;
   using Graph = GenericGraph<Node>;
 
   Graph TypeGraph;
-  std::map<const model::Type *, Node *> TypeToNode;
+  std::map<const model::TypeDefinition *, Node *> TypeToNode;
 
-  llvm::SmallPtrSet<Type *, 16> ToKeep;
+  llvm::SmallPtrSet<const model::TypeDefinition *, 16> ToKeep;
```

Comment:
Why are you adding `const`?

Reply:
Ah, you were talking about this part? Then I guess I got confused because I wrongly assumed which commit last comment was attached to.

> Why are you adding `const`? Below too.

Short answer is that it doesn't matter: these are only used a lookup tables.

The long answer is that at the point it was written `TypeDefinition::edges` was const-only - hence I was getting compilation errors. Sadly, I didn't manage to keep it that way: so there's no need to add it anymore. Either way works for me.